proposals/lip-0045.md

```diff
@@ -1575,9 +1492,82 @@ genesisInteroperabilityStoreSchema = {
 
 Here, the `...` notation, borrowed from [JavaScript ES6 data destructuring](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Destructuring_assignment#Object_destructuring), indicates that the corresponding schema should be inserted in place, and it is just used for notational convenience.
 
+#### Genesis State Initialization
+
+During the genesis state initialization stage of a genesis block `g`, the following steps are executed. If any step fails, the block is discarded and has no further effect.
+
+Let `genesisBlockAssetBytes` be the `data` bytes included in the genesis block assets for the Interoperability module and let `interoperabilityAsset = decode(genesisInteroperabilityStoreSchema, genesisBlockAssetBytes)`. Let `ownChainName = interoperabilityAsset.ownChainName`, `ownChainNonce = interoperabilityAsset.ownChainNonce`, `chainInfos = interoperabilityAsset.chainInfos`, `terminatedStateAccounts = interoperabilityAsset.terminatedStateAccounts`, and `terminatedOutboxAccounts = interoperabilityAsset.terminatedOutboxAccounts`.
+
+##### Genesis Asset Verification
+
+The genesis asset verification follows different rules on the mainchain or on a sidechain.
+
+On the mainchain:
+
+* `ownChainName == CHAIN_NAME_MAINCHAIN`.
+* If `chainInfos` is non-empty, then `ownChainNonce > 0`. Conversely, if `chainInfos` is empty, then `ownChainNonce == 0`.
+* Each entry `chainInfo` in `chainInfos` has a unique `chainInfo.chainID` and is ordered lexicographically by `chainInfo.chainID`. Furthermore for each entry it holds:
+  * `chainInfo.chainID != getMainchainID()`;
+  * `chainInfo.chainId[0] == getMainchainID()[0]`.
+* For each entry `chainInfo` in `chainInfos`, let `chainData = chainInfo.chainData`, then check:
+  * `chainData.lastCertificate.timestamp < g.header.timestamp`;
+  * all `chainData.name` are unique;
+  * all `chainData.name` are from the character set `a-z0-9!@$&_.` and length between `MIN_CHAIN_NAME_LENGTH` and `MAX_CHAIN_NAME_LENGTH`;
+  * property `chainData.status` is in set `{CHAIN_STATUS_REGISTERED, CHAIN_STATUS_ACTIVE, CHAIN_STATUS_TERMINATED}`.
+* For each entry `chainInfo` in `chainInfos`, let `channelData = chainInfo.channelData`, then check:
+  * `channelData.messageFeeTokenID == Token.getTokenIDLSK()`.
+* For each entry `chainInfo` in `chainInfos`, let `activeValidators = chainInfo.chainValidators.activeValidators` and let `certificateThreshold = chainInfo.chainValidators.certificateThreshold`, then check:
+  * `activeValidators` must have at least 1 element and at most `MAX_NUM_VALIDATORS` elements;
+  * `activeValidators` must be ordered lexicographically by `blsKey` property;
+  * all `blsKey` properties must be pairwise distinct;  
+  * let `totalWeight` be the sum of the `bftWeight` property of every element in `activeValidators`. Then `totalWeight` has to be less than or equal to `MAX_UINT64`;
```

Comment:
We must also check `bftWeight > 0`.

Reply:
added

Reply:
I would write " be the sum of the `bftWeight` property of all elements in `activeValidators`"

Reply:
Same for sidechains.